Monika After Story/game/script-apologies.rpy

```diff
@@ -0,0 +1,260 @@
+                m 3eka "We just have to accept what happened, learn from our mistakes, and move on, together. Okay?"
+            elif mas_isMoniNormal(higher=True):
+                m 1eka "[player]..."
+                m "If you want to apologize, go ahead. It'd mean a lot to me if you did."
+            elif mas_isMoniUpset():
+                m 2rkc "Oh..."
+                m "I was kind of--"
+                $ _history_list.pop()
+                m 2dkc "Nevermind."
+            elif mas_isMoniDis():
+                m 6rkc "...?"
+            else:
+                m 6ckc "..."
+        else:
+            if mas_isMoniAff(higher=True):
+                m 1hua "Ehehe, you don't have anything to be sorry about silly~"
+            elif mas_isMoniNormal(higher=True):
+                m 1eka "I'm not sure what you wanted to apologize for, but I'm sure that it was nothing."
+            elif mas_isMoniDis(higher=True):
+                m 1rkc "Did you have something to say, [player]?"
+            else:
+                m "..."
+        return
+
+
+    #Call our apology label
+    #NOTE: mas_setApologyReason() ensures that this label exists
+    call expression apology
+
+    #Increment the shown count
+    $ mas_getEV(apology).shown_count += 1
+
+    #Lock the apology label if it's not the generic
+    if apology != "mas_apology_generic":
+        $ store.mas_lockEVL(apology, 'APL')
+
+    #Pop that apology from the time db
+    if apology in persistent._mas_apology_time_db: #sanity check
+        $ persistent._mas_apology_time_db.pop(apology)
+    return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent._mas_apology_database,
+            prompt="...for something else.",
+            eventlabel="mas_apology_generic",
+            unlocked=True,
+        ),
+        code="APL"
+    )
+
+label mas_apology_generic:
+    #dict of all generic apologies
+    #Note, if a custom apology is needed, add it here and reference the apology reason by the integer associated.
+    $ mas_apology_reason_db = {
+        0: "",
+        1: "saying you wanted to break up. I knew you didn't mean it...",
+        2: "joking about having another girlfriend. You really gave me a heart attack!",
+        3: "calling me a murderer. I hope you don't really see me that way...",
+        4: "closing the game on me.",
+        5: "entering my room without knocking.",
+        6: "missing Christmas.",
+        7: "forgetting my birthday.",
+        8: "not spending time with me on my birthday.",
+        9: "the game crashing. I understand it happens sometimes, but don't worry, I'm alright!",
+        10: "the game crashing. It really was scary, but I'm just glad you came back to me and made things better."
+    }
+
+    #We want to make sure that the None possibility is still a thing after apologizing once.
+    if mas_apology_reason is not None:
+        $ apology_reason = mas_apology_reason_db.get(mas_apology_reason,mas_apology_reason_db[0])
```

Comment:
it looks like you only use `apology_reason` in the `elif mas_apology_reason` so you should just move this line to into that elif block. you wont have to check for not None.